internal/engine/bundle_builder.go

```diff
@@ -69,23 +72,44 @@ func (b *BundleBuilder) InitWorkspace(workspace string) error {
 	var files []string
 	for i, file := range b.files {
 		_, fn := filepath.Split(file)
-		dstFile := filepath.Join(workspace, fmt.Sprintf("%v.%s", i, fn))
-		files = append(files, dstFile)
-		if err := cp.Copy(file, dstFile); err != nil {
-			return err
-		}
-	}
-
-	for _, f := range files {
-		_, fn := filepath.Split(f)
-		data, err := b.injector.Inject(f)
+		dstFile := filepath.Join(workspace, fmt.Sprintf("%v.%s.cue", i, fn))
+		f, err := os.Open(file)
 		if err != nil {
-			return fmt.Errorf("failed to inject %s: %w", fn, err)
+			return err
 		}
-
-		if err := os.WriteFile(f, data, os.ModePerm); err != nil {
-			return fmt.Errorf("failed to inject %s: %w", fn, err)
+		defer f.Close()
+
+		if a, err := yaml.Extract(file, f); err == nil {
```

Comment:
This approach has many issues, please let's look at the file extension in a switch statement instead of hiding errors. If a cue, json or yaml is invalid, we should tell users and exit, instead of silently ignoring errors which would result in applying something on a cluster that was never intended.

Reply:
Okay with me, but it means we don't support file with incorrect extension (which is probably a fair assumption).

Reply:
Yes we don't support incorrect extension, the switch should match:  https://github.com/stefanprodan/timoni/blob/main/cmd/timoni/build.go#L205

Reply:
Perfect, going to grab some food and pushing you a fix after

Reply:
Thanks

Reply:
Okay did a switch around extensions. Also refactored the injector so it harmonizes the pattern and avoid double parsing / writing of files. I also dropped the copy dependency in the meantime.

Reply:
Looks great, thanks a lot @b4nst.

I was wandering if we can create the workspace in-memory instead of on disk in tmp. I haven't figure out how to pass bytes to cue build, but it must be a way.

Reply:
New PR for sure, after we get this in shape and merged. Thanks for looking into it.

Reply:
We might want to revive https://github.com/cue-lang/cue/issues/607 for a proper in-memory support